modules/signals/src/deep-computed.ts

```diff
@@ -0,0 +1,8 @@
+import { computed } from '@angular/core';
+import { DeepSignal, toDeepSignal } from './deep-signal';
+
+export function deepComputed<T extends object>(
```

Comment:
@markostanimirovic what's the reasoning to restrict `T` to be an object if `toDeepSignal` supports "any"?
sometimes we have selectors that returns primitives or objects depending on the context :thinking:

Reply:
Yes, that's a requirement. `deepComputed` uses the initial value to create a proxy for deeply nested signals.

Reply:
@markostanimirovic it's basically an alias of `toDeepSignal` which doesn't have that requirement

Reply:
If a signal with a primitive value is provided to the `toDeepSignal` function, it's just returned without further processing. Check the implementation: https://github.com/ngrx/platform/blob/main/modules/signals/src/deep-signal.ts#L26-L29

If a primitive type or union that contains primitive is passed to the `DeepSignal` type, it will be resolved to `Signal`:

```ts
import { DeepSignal } from '@ngrx/signals';

declare const s1: DeepSignal<string>; // type: Signal<string>
declare const s2: DeepSignal<{ foo: string } | string>; // type: Signal<{ foo: string } | string>
```

I could remove the `extends object` restriction from the `deepComputed` input, but then, users may be confused why the return type is resolved to `Signal`:

```ts
const s1 = deepComputed(() => 1 as number | { foo: string }); // type: Signal<number | { foo: string }>
```

In other words, there is no difference in this example whether you use `computed` or `deepComputed`.

With the `extends object` restriction it's clear - use `deepComputed` for objects, otherwise, it has no impact and you can go with `computed`.